ui/guest/src/react/GuestProxy.tsx

```diff
@@ -196,23 +199,22 @@ export function GuestProxy() {
         document.querySelectorAll<HTMLElement>('[data-craftercms-model-id]').forEach(registerElement);
       });
 
-    const handler: JQuery.EventHandlerBase<any, any> = (e: Event): void => {
-      let record = ElementRegistry.fromElement(e.currentTarget as Element);
+    const handler: JQuery.EventHandlerBase<any, any> = (e: Event, target): void => {
+      let record = ElementRegistry.fromElement(target as Element);
       if (notNullOrUndefined(record)) {
         persistenceRef.current.onEvent(e, record.id);
       }
     };
 
-    $(document)
-      .on('mouseover', '[data-craftercms-model-id]', handler)
-      .on('mouseleave', '[data-craftercms-model-id]', handler)
-      .on('dragstart', '[data-craftercms-model-id]', handler)
-      .on('dragover', '[data-craftercms-model-id]', handler)
-      .on('dragleave', '[data-craftercms-model-id]', handler)
-      .on('drop', '[data-craftercms-model-id]', handler)
-      .on('dragend', '[data-craftercms-model-id]', handler)
-      .on('click', '[data-craftercms-model-id]', handler)
-      .on('dblclick', '[data-craftercms-model-id]', handler);
+    delegateEventListener(document, 'mouseover', '[data-craftercms-model-id]', handler);
+    delegateEventListener(document, 'mouseout', '[data-craftercms-model-id]', handler);
```

Comment:
The javascript approach uses event bubbling, so we need to use `mouseout` instead of `mouseleave` since mouseleave does not bubble. `mouseout` is fired when the pointer leaves the element *or* leaves of the element's descendants, and taking that into consideration helps achieving event delegation

Reply:
Can we identify how event abstraction libraries (jQuery, React, etc) do it? I worry how this might affect things.